### PR TITLE
K8s: Implement playlist api with k8s client

### PR DESCRIPTION
**Pull request by ryantxu**
This PR adds a feature flag that will change requests to `/api/playlists` so that it is implemented with k8s requests rather than calling a service directly.  This puts us on a path to be able to know that all external requests go through the k8s service rather than just some.

### CODE REVIEW

File: pkg/api/playlist.go [hunk 1]
```diff
@@ -2,15 +2,74 @@ package api
...
+					continue // query filter
+				}
+				playlists = append(playlists, *p)
+			}
+			return response.JSON(http.StatusOK, playlists)
+		}
+	}
+
+	// Playlist API
+	apiRoute.Group("/playlists", func(playlistRoute routing.RouteRegister) {
```
Comment (ryantxu) [inline pkg/api/playlist.go:62]: After https://github.com/grafana/grafana/pull/75709 -- we can replace all endpoints

File: pkg/kinds/general.go [hunk 2]
```diff
@@ -40,23 +40,36 @@ type GrafanaResource[Spec any, Status any] struct {
...
-const annoKeyOriginPath = "grafana.com/originPath"
-const annoKeyOriginKey = "grafana.com/originKey"
-const annoKeyOriginTime = "grafana.com/originTime"
+const annoKeyOriginName = "grafana.app/originName"
+const annoKeyOriginPath = "grafana.app/originPath"
+const annoKeyOriginKey = "grafana.app/originKey"
+const annoKeyOriginTimestamp = "grafana.app/originTimestamp"
+
+func (m *GrafanaResourceMetadata) set(key string, val string) {
+	if val == "" {
```
Comment (ryantxu) [inline pkg/kinds/general.go:62]: mostly part of https://github.com/grafana/grafana/pull/77401

File: pkg/apis/playlist/v0alpha1/conversions.go [hunk 3]
```diff
@@ -22,14 +36,38 @@ func convertToK8sResource(v *playlist.PlaylistDTO, namespacer request.NamespaceM
...
 			Namespace:         namespacer(v.OrgID),
+			Annotations:       meta.Annotations,
 		},
 		Spec: spec,
 	}
 }
+
+// Convert Legacy ID to
+func getLegacyID(item *unstructured.Unstructured) int64 {
+	s, ok := item.GetAnnotations()["grafana.com/originKey"]
```
Comment (DanCech) [inline pkg/apis/playlist/v0alpha1/conversions.go:65]: should use an accessor here vs the hardcoded name

File: pkg/services/playlist/playlistimpl/playlist.go [hunk 4]
```diff
@@ -68,6 +68,7 @@ func (s *Service) Get(ctx context.Context, q *playlist.GetPlaylistByUidQuery) (*
 		}
 	}
 	return &playlist.PlaylistDTO{
+		Id:        v.Id,
```
Comment (ryantxu) [inline pkg/services/playlist/playlistimpl/playlist.go:71]: added so we can include it as origin info

**Review by DanCech — APPROVED** [review]
LGTM

**Pull request merged**